src/runtime/objmodel.cpp

```diff
@@ -6479,7 +6479,9 @@ extern "C" void setitem(Box* target, Box* slice, Box* value) {
             RewriterVar* r_cls = r_obj->getAttr(offsetof(Box, cls));
             RewriterVar* r_m = r_cls->getAttr(offsetof(BoxedClass, tp_as_mapping));
             r_m->addGuardNotEq(0);
-            rewriter->call(true, (void*)setitemHelper, r_obj, r_slice, r_value);
+            r_m->addAttrGuard(offsetof(PyMappingMethods, mp_ass_subscript), (intptr_t)m->mp_ass_subscript);
+            RewriterVar* r_ret = rewriter->call(true, (void*)m->mp_ass_subscript, r_obj, r_slice, r_value);
+            rewriter->checkAndThrowCAPIException(r_ret, -1, assembler::MovType::L);
             rewriter->commit();
         }
 
```

Comment:
could you please also remove the `setitemHelper` function and replace the call with
```
int ret = m->mp_ass_subscript(target, slice, value);
if (ret == -1)
   throwCAPIException();
```